src/math/StakeMath.sol

```diff
@@ -0,0 +1,189 @@
+     * specified duration
+     * @param _balance Account current balance
+     * @param _currentMaxMP Account current max multiplier points
+     * @param _currentLockEndTime Account current lock end timestamp
+     * @param _processTime Process current timestamp
+     * @param _increasedAmount Increased amount of balance
+     * @param _increasedLockSeconds Increased amount of seconds to lock
+     * @return _deltaMpTotal Increased amount of total multiplier points
+     * @return _deltaMpMax Increased amount of max multiplier points
+     * @return _newLockEnd Account new lock end timestamp
+     */
+    function _calculateStake(
+        uint256 _balance,
+        uint256 _currentMaxMP,
+        uint256 _currentLockEndTime,
+        uint256 _processTime,
+        uint256 _increasedAmount,
+        uint256 _increasedLockSeconds
+    )
+        internal
+        pure
+        returns (uint256 _deltaMpTotal, uint256 _deltaMpMax, uint256 _newLockEnd)
+    {
+        uint256 newBalance = _balance + _increasedAmount;
+        if (newBalance < MIN_BALANCE) {
+            revert StakeManager__StakeIsTooLow();
+        }
+        _newLockEnd = Math.max(_currentLockEndTime, _processTime) + _increasedLockSeconds;
+        uint256 dt_lock = _newLockEnd - _processTime;
+        if (dt_lock != 0 && (dt_lock < MIN_LOCKUP_PERIOD || dt_lock > MAX_LOCKUP_PERIOD)) {
+            revert StakeManager__InvalidLockTime();
+        }
+
+        uint256 deltaMpBonus;
+        if (dt_lock > 0) {
+            deltaMpBonus = _bonusMP(_increasedAmount, dt_lock);
+        }
+
+        if (_balance > 0 && _increasedLockSeconds > 0) {
+            deltaMpBonus += _bonusMP(_balance, _increasedLockSeconds);
+        }
+
+        _deltaMpTotal = _initialMP(_increasedAmount) + deltaMpBonus;
+        _deltaMpMax = _deltaMpTotal + _accrueMP(_increasedAmount, MAX_MULTIPLIER * YEAR);
+
+        if (_deltaMpMax + _currentMaxMP > MP_MPY_ABSOLUTE * newBalance) {
+            revert StakeManager__AbsoluteMaxMPOverflow();
+        }
+    }
+
+    /**
+     * @notice Calculates the bonus multiplier points earned when a balance Δa is locked for a specified duration
+     * @param _balance Account current balance
+     * @param _currentMaxMP Account current max multiplier points
+     * @param _currentLockEndTime Account current lock end timestamp
+     * @param _processTime Process current timestamp
+     * @param _increasedLockSeconds Increased amount of seconds to lock
+     * @return _deltaMp Increased amount of total and max multiplier points
+     * @return _newLockEnd Account new lock end timestamp
+     */
+    function _calculateLock(
+        uint256 _balance,
+        uint256 _currentMaxMP,
+        uint256 _currentLockEndTime,
+        uint256 _processTime,
+        uint256 _increasedLockSeconds
+    )
+        internal
+        pure
+        returns (uint256 _deltaMp, uint256 _newLockEnd)
+    {
+        require(_balance > 0);
+        require(_increasedLockSeconds > 0);
```

Comment:
These should probably be equivalent to the checks done here: https://github.com/vacp2p/staking-reward-streamer/pull/109/files#diff-b8371c7492fb939a38fb61c84fb63d70e61f70c633e1a09e6e059c8372ec3846L241-L253

So I guess it's kinda the same as the comment I made in `stake()` regarding error handling.
At the very least, these should throw custom errors.